Program.cs

```diff
@@ -253,6 +253,54 @@ static void CityMenu(Statistics cityStats)
                         Console.WriteLine("Press any key to continue");
                         Console.ReadKey();
                         break;
+                    case "7":
+                        bool valid = false;
+                        string fileName = "";
+                        bool isULong;
+                        ulong newPopulation = 0;
+                        Console.WriteLine("Available data files:\n");
+                        Console.WriteLine("\t1) Canadacities-JSON.json");
+                        Console.WriteLine("\t2) Canadacities-XML.xml");
+                        Console.WriteLine("\t3) Canadacities.csv");
+                        Console.Write("Please make a selection(ex. 1, 2): ");
+                        do
+                        {
+                            selection = Console.ReadLine();
+                            switch (selection.ToLower())
+                            {
+                                case "1":
+                                    fileName = "Canadacities-JSON.json";
+                                    valid = true;
+                                    break;
+                                case "2":
+                                    fileName = "Canadacities-XML.xml";
+                                    valid = true;
+                                    break;
+                                case "3":
+                                    fileName = "Canadacities.csv";
+                                    valid = true;
+                                    break;
+                                default:
+                                    Console.Write("\nInvalid selection, please enter a valid selection: ");
+                                    break;
+                            }
+                        } while (!valid);
+                        Console.WriteLine("\nWhich city do you want to change the population?\n");
+                        city = GetCityChoice(cityStats);
+                        string oldPopulation = string.Format("{0:n0}", city.GetPopulation());
+                        Console.WriteLine($"\nThe city of{city.GetCityName()}, {city.GetProvince()} has a current population of {oldPopulation}.");
+                        do
+                        {
+                            Console.Write($"Please enter the new population for {city.GetCityName()}, {city.GetProvince()} :");
+                            isULong = UInt64.TryParse(Console.ReadLine(), out newPopulation);
+                            if (!isULong)
+                                Console.Write("Invalid response. ");
+                        }
+                        while (!isULong);
+                        cityStats.UpdatePopulation($"{city.GetCityName()}|{city.GetProvince()}", newPopulation, fileName);
```

Comment:
Make sure to put this in a try-catch, the method throws exceptions

Reply:
Also take a look at the UpdatePopulation method, it accesses cities by the old way, we need to update it to use getSpecificCity

Reply:
Bonus points of you comment that method comment for it